airflow/models.py

```diff
@@ -1407,11 +1407,15 @@ def signal_handler(signum, frame):
 
                 # If a timeout is specified for the task, make it fail
                 # if it goes beyond
+                task_timeout = task_copy.execution_timeout
+                if hasattr(task_copy, 'timeout') and task_copy.timeout:
```

Comment:
isn't just `if hasattr(task_copy, 'timeout')` enough?

Reply:
is `task_copy.timeout` default to None for `BaseSensor`?

Reply:
Sensor has timeout parameter, but if can be none which means we should use execution_timeout instead

Reply:
@milton0825 No. I have a PR out to change LyftExternalTaskSensor https://github.com/lyft/python-lyft-etl/pull/226

Reply:
+1

Reply:
Actually, maybe just changing LyftExternalTaskSensor will be enough. Holding off this PR.